locked-asset/lkmex-transfer/src/lib.rs

```diff
@@ -55,6 +59,43 @@ pub trait LkmexTransfer:
         self.add_energy_to_destination(receiver, &funds);
     }
 
+    #[endpoint]
+    fn cancel_transfer(&self, sender: ManagedAddress, receiver: ManagedAddress) {
+        self.require_caller_has_admin_permissions();
+        let locked_funds_mapper = self.locked_funds(&receiver, &sender);
+        require!(!locked_funds_mapper.is_empty(), TRANSFER_NON_EXISTENT);
+
+        let locked_funds = locked_funds_mapper.get();
+        locked_funds_mapper.clear();
```

Comment:
There's already a method for SingleValueMapper which does both get and clear:
```suggestion
        let locked_funds = locked_funds_mapper.take();
```

Reply:
The framework version is too old.